ci-operator/step-registry/assisted/vsphere/postinstall/assisted-vsphere-postinstall-commands.sh

```diff
@@ -34,26 +93,25 @@ EOF
 
 oc patch kubecontrollermanager cluster -p='{"spec": {"forceRedeploymentReason": "recovery-'"$( date --rfc-3339=ns )"'"}}' --type=merge
 
-cat <<EOF | oc apply -f -
-apiVersion: v1
-kind: ConfigMap
-metadata:
-  name: cloud-provider-config
-  namespace: openshift-config
-data:
-  config: |
-    [Global]
-    secret-name = "vsphere-creds"
-    secret-namespace = "kube-system"
-    insecure-flag = "1"
-    [Workspace]
-    server = "${VSPHERE_VCENTER}"
-    datacenter = "${VSPHERE_DATACENTER}"
-    default-datastore = "${VSPHERE_DATASTORE}"
-    folder = "${VSPHERE_FOLDER}"
-    [VirtualCenter "${VSPHERE_VCENTER}"]
-    datacenters = "${VSPHERE_DATACENTER}"
-EOF
+# Do the following if OCP version is >=4.13
+compare_versions "$version" "4.12"
+if [[ $? -eq 0 ]]; then
+    # Taint the nodes with the uninitialized taint
+    nodes=$(oc get nodes -o wide | awk '{print $1}' | tail -n +2)
+    for NODE in $nodes; do
+      oc adm taint node "$NODE" node.cloudprovider.kubernetes.io/uninitialized=true:NoSchedule
```

Comment:
shouldn't it be tainted after the config is updated? I guess the taint allows the controller to reload the configuration and get the right credentials?

Reply:
Nope, before updating the infrastructure config

Reply:
what is the reason?